src/commands.ts

```diff
@@ -694,7 +722,18 @@ export const insertSection =
 
     let pos
     if (hasParentNodeOfType(schema.nodes.body)(selection) || subsection) {
-      pos = findPosAfterParentSection(state.selection.$from)
+      /*
+      Determine the position where the new section or subsection will be inserted.
+      If we are inserting a subsection (`subsection` is true):
+        - First, try to find the position before the first existing subsection in the current section using `findPosBeforeFirstSubsection`.
+        - If there are no existing subsections (or the function returns null), fallback to finding the position immediately after the parent section using `findPosAfterParentSection`.
+      If we are inserting a regular section (`subsection` is false):
+        - Simply find the position after the parent section using `findPosAfterParentSection`.
+      */
+      pos = subsection
+        ? findPosBeforeFirstSubsection(state.selection.$from) ||
+          findPosAfterParentSection(state.selection.$from)
```

Comment:
looks like it can be done once in advance to avoid duplicated execution in the tertiary operator.

Reply:
No. We search for the subsection position only if we insert a subsection.

Reply:
Got it. Thanks!